src/ResourceManagement/PrivateDns/VirtualNetworkLink/VirtualNetworkLinkImpl.cs

```diff
@@ -0,0 +1,489 @@
+
+        /// <summary>
+        /// Gets the status of the virtual network link to the Private DNS zone.
+        /// </summary>
+        public VirtualNetworkLinkState VirtualNetworkLinkState
+        {
+            get
+            {
+                return Inner.VirtualNetworkLinkState;
+            }
+        }
+
+        /// <summary>
+        /// Gets the provisioning state of the virtual network link.
+        /// </summary>
+        public ProvisioningState ProvisioningState
+        {
+            get
+            {
+                return Inner.ProvisioningState;
+            }
+        }
+
+        /// <summary>
+        /// Gets resource ID of the record set.
+        /// </summary>
+        public string Id
+        {
+            get
+            {
+                return Inner.Id;
+            }
+        }
+
+        /// <summary>
+        /// Attaches the child definition to the parent resource definiton.
+        /// </summary>
+        /// <return>The next stage of the parent definition.</return>
+        PrivateDnsZone.Definition.IWithCreate ResourceManager.Fluent.Core.ChildResource.Definition.IInDefinition<PrivateDnsZone.Definition.IWithCreate>.Attach()
+        {
+            return Parent;
+        }
+
+        public override async Task<IVirtualNetworkLink> CreateAsync(CancellationToken cancellationToken)
+        {
+            return await CreateOrUpdateAsync(Inner, cancellationToken);
+        }
+
+        public override async Task DeleteAsync(CancellationToken cancellationToken)
+        {
+            await Parent.Manager.Inner.VirtualNetworkLinks.DeleteAsync(
+                Parent.ResourceGroupName,
+                Parent.Name,
+                Name(),
+                ifMatch: eTagState.IfMatchValueOnDelete(),
+                cancellationToken: cancellationToken);
+        }
+
+        /// <summary>
+        /// Specifies to disable auto-registration for virtual network records.
+        /// </summary>
+        /// <return>The next stage of the definition.</return>
+        VirtualNetworkLink.Definition.IWithAttach<PrivateDnsZone.Definition.IWithCreate> VirtualNetworkLink.Definition.IWithAutoRegistration<PrivateDnsZone.Definition.IWithCreate>.DisableAutoRegistration()
+        {
+            return DisableAutoRegistrationInternal();
+        }
+
+        /// <summary>
+        /// Specifies to enable auto-registration for virtual network records.
+        /// </summary>
+        /// <return>The next stage of the definition.</return>
+        VirtualNetworkLink.Definition.IWithAttach<PrivateDnsZone.Definition.IWithCreate> VirtualNetworkLink.Definition.IWithAutoRegistration<PrivateDnsZone.Definition.IWithCreate>.EnableAutoRegistration()
+        {
+            return EnableAutoRegistrationInternal();
+        }
+
+        public override async Task<IVirtualNetworkLink> UpdateAsync(CancellationToken cancellationToken)
+        {
+            var resource = await Parent.Manager.Inner.VirtualNetworkLinks.GetAsync(
```

Comment:
Why get a new resource?

Reply:
We get a existing resource here, and copy local changes into it, then pass it to CreateOrUpdate call. So it will be update operation here.

Reply:
Typical flow is just update current Inner, and send the PUT or PATCH. The question is this: why do an extra GET?

Reply:
Typically we just update current Inner, but for updating children level, we need to have extra GET to get full information of the child to update